### PR TITLE
studio: unsupported file detection

### PR DESCRIPTION
**Pull request by achen5671**
## Summary
https://tsu.atlassian.net/browse/IR-4594
## References
closes #_insert number here_

### CODE REVIEW

File: packages/editor/src/panels/files/toolbar.tsx [hunk 1]
```diff
@@ -178,117 +181,156 @@ export default function FilesToolbar() {
 
   const { backDirectory, refreshDirectory, createNewFolder } = useCurrentFiles()
 
+  const UnsupportedFileModal = ({ message }) => {
+    return (
+      <Modal
+        onSubmit={() => {
+          return true
```
Comment (aditya-mitra) [inline packages/editor/src/panels/files/toolbar.tsx]: why do we have a return here?

File: packages/editor/src/panels/files/toolbar.tsx [hunk 2]
```diff
@@ -178,117 +181,156 @@ export default function FilesToolbar() {
...
+              .catch((err) => {
+                PopoverState.showPopupover(<UnsupportedFileModal message={err.message} />)
+                NotificationService.dispatchNotify(err.message, { variant: 'error' })
+              })
+          }
+        >
+          {t('editor:layout.filebrowser.uploadFiles')}
+        </Button>
+        <Button
+          id="uploadFiles"
```
Comment (aditya-mitra) [inline packages/editor/src/panels/files/toolbar.tsx]: also remove this id

File: packages/editor/src/panels/files/toolbar.tsx [hunk 3]
```diff
@@ -178,117 +181,156 @@ export default function FilesToolbar() {
...
+        <ViewModeSettings />
 
-      <ViewModeSettings />
+        <div className="w-30 flex h-7 flex-row items-center gap-1 rounded bg-[#2F3137] px-1 py-1 ">
+          {VIEW_MODES.map(({ mode, icon }) => (
+            <Button
+              key={mode}
+              variant="transparent"
+              startIcon={icon}
+              className={`p-0 ${filesViewMode.value !== mode ? 'opacity-50' : ''}`}
```
Comment (aditya-mitra) [inline packages/editor/src/panels/files/toolbar.tsx:271]: use twmerge here

**Review by aditya-mitra — DISMISSED** [review]

**Review by JT00y — APPROVED** [review]
looks good

File: packages/editor/src/functions/assetFunctions.ts [hunk 4]
```diff
@@ -35,6 +35,56 @@ import { pathJoin } from '@ir-engine/engine/src/assets/functions/miscUtils'
...
+}
+
+const supportedFiles = {
+  [FileType.THREE_D]: new Set(['GLTF', 'GLB']),
+  [FileType.IMAGE]: new Set(['.png', '.tiff', '.jpg', '.jpeg', '.gif', '.ktx2']),
+  [FileType.AUDIO]: new Set(['.mp3', '.mpeg', '.m4a', '.wav']),
+  [FileType.VIDEO]: new Set(['.mp4', '.mkv', '.avi'])
+}
+
+function findMimeType(file): FileType {
```
Comment (JT00y) [inline packages/editor/src/functions/assetFunctions.ts:61]: this part don't have condition for FileType.THREE_D, not a big issue

**Pull request merged**